packages/atoms/connect/ConnectButton.tsx

```diff
@@ -13,44 +13,40 @@ import type { AtomsGlobalConfigProps } from "../types";
 interface ConnectButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
   buttonText?: string;
   icon?: JSX.Element;
-  handleClick: () => void;
   onSuccess?: () => void;
   onError: () => void;
 }
 
 export function ConnectButton({
   buttonText,
-  handleClick,
+  onClick,
   onSuccess,
   onError,
   className,
   icon,
 }: ConnectButtonProps & AtomsGlobalConfigProps) {
   const [isProcessing, setIsProcessing] = useState<boolean>(false);
-  const [isSuccess, setIsSuccess] = useState<boolean>(false);
   const [errMsg, setErrMsg] = useState<string>("");
 
   function handleSubmit(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
     e.preventDefault();
 
     try {
       setIsProcessing(true);
-      setIsSuccess(true);
-      handleClick();
+      onClick && onClick(e);
+
+      // if user wants to handle onSuccess inside handleClick then it makes no sense to have a separate handler
+      // otherwise only if the user explicitly passes an onSuccess handler this gets triggered
+      if (onSuccess) {
```

Comment:
Since onSuccess can be handled inside onClick if a user decides to do so that's why the if statement to check if the user passes a value for onSucess